source/grid/reference_cell.cc

```diff
@@ -807,6 +807,256 @@ ReferenceCell::gmsh_element_type() const
 
 
 
+namespace
+{
+  double
+  distance_to_line_square(const Point<1> &x0,
+                          const Point<1> &x1,
+                          const Point<1> &p)
+  {
+    return std::min(x0.distance_square(p), x1.distance_square(p));
+  }
+
+
+
+  double
+  distance_to_line_square(const Point<2> &x0,
+                          const Point<2> &x1,
+                          const Point<2> &p)
+  {
+    const double t =
+      -(x1[0] * x0[0] - std::pow(x0[0], 2) - (x1[0] - x0[0]) * p[0] +
+        x1[1] * x0[1] - std::pow(x0[1], 2) - (x1[1] - x0[1]) * p[1]) /
+      (std::pow(x1[0], 2) - 2 * x1[0] * x0[0] + std::pow(x0[0], 2) +
+       std::pow(x1[1], 2) - 2 * x1[1] * x0[1] + std::pow(x0[1], 2));
```

Comment:
This is hard to read. Couldn't this be written in the form of some dot products etc? If so, it may also cover the 3d case. I don't see how 2d/3d needs to be treated any different.

Separately, can you add a comment what `t` refers to?

Reply:
Fixed!